components/charts/SpendByCategoryChart.tsx

```
@@ -74,61 +74,132 @@ export const SpendByCategory = withObservables<
+        categoryName: categories.find(c => c.id === category)?.name || 'Uncategorized',
+      }))
+      .sort((a, b) => b.total - a.total);
+
+    // Filter out categories with zero spending
+    const nonZeroData = dataWithNames.filter(item => item.total > 0);
+    
+    return nonZeroData.slice(0, maxCategories);
+  }, [chartData, categories, maxCategories]);
+
+  // Dynamic label formatting based on number of categories and screen size
+  const formatCategoryLabel = React.useCallback((categoryName: any) => {
```

ESLint

[error] Insert `⏎····`

(prettier/prettier)